### ISSUE TITLE
Use redfish-virtualmedia without DHCP server

### ISSUE THREAD

**Issue by mel1nn** [metal3-io/baremetal-operator#731]
Hi, I want to use the redfish-virtualmedia process for my baremetal servers and I want to know if there is a way to inject network configuration in the kernel image instead of using the DHCP server of Ironic. It seems that Ironic enables the network configuration injection with virtual media (https://docs.openstack.org/ironic/latest/admin/drivers/redfish.html#layer-3-or-dhcp-less-ramdisk-booting). So, do you plan to add the feature of injecting network configuration into the kernel image before the ironic-python-agent starts ?
Thanks a lot !

**Comment by levsha** [comment]
Another solution could be to use out-of-band inspection that redfish supports: https://docs.openstack.org/ironic/latest/admin/drivers/redfish.html#out-of-band-inspection
I've tried some dirty change and it seems to be doable: https://github.com/levsha/baremetal-operator/tree/no-inspector
IPA will be still required to provision the image, but solution for that step could be to use live-iso format.

**Comment by dtantsur** [comment]
/remove-lifecycle stale

We're still interested in DHCP-less deploy, I'm working on certain blockers in RHEL/CentOS 8 support upstream.

**Comment by levsha** [comment]
I was able to boot Fedora CoreOS live iso using idrac-virtualmedia in L3 DHCP-less network: https://gitlab.com/iatlevsha/ironic/-/tree/master/deployments/kubernetes/ironic/idrac-l3
Still WIP.

Notable steps I've done for this:
* built my own minimal ironic image (might not be required, but I've found it easier to do ironic configuration changes another way)
* removed the line in BMO code that explicitly sets "inspector" inspect interface - rely on ironic configuration
* enabled Out-Of-Band iDRAC inspection on ironic side
* removed all direct calls to ironic inspector from BMO, filled up (some) node information from ironic instead
* disabled automated clean
* deleted raid interface for idrac-virtualmedia in BMO to not trigger manual clean that tries to delete RAID configuration (might not be required)
* used "live-iso" image.

All these changes removed need of Ironic Python Agent ram image: Ironic uses iDRAC only to inspect the host and attaches iso image as Virtual Media then.
There is currently no way to dynamically inject any userData/networkData in "live-iso" mode. Therefore I plan to build iso image per host with ignition config already injected that will also contain all required networking configuration.
